gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTable.java

```diff
@@ -86,6 +96,117 @@ public KnoxShellTable value(Comparable<? extends Object> value) {
     return col;
   }
 
+  public Conversions getConversion(Comparable<? extends Object> colIndex) {
+      Conversions type = null;
+
+      if (colIndex instanceof Double) {
+        type = Conversions.DOUBLE;
+      }
+      else if (colIndex instanceof Integer) {
+        type = Conversions.INTEGER;
+      }
+      else if (colIndex instanceof Float) {
+        type = Conversions.FLOAT;
+      }
+      else if (colIndex instanceof Byte) {
+        type = Conversions.BYTE;
+      }
+      else if (colIndex instanceof Short) {
+        type = Conversions.SHORT;
+      }
+      else if (colIndex instanceof Long) {
+        type = Conversions.LONG;
+      }
+      return type;
```

Comment:
Should have some default case?

Reply:
I'm not entirely sure a default case is needed since the updated patch now has an else that throws an `IllegalArgumentException`, as per the response to the review below.